### DISCUSSION TITLE
jv_snapshots collection saving Date data type field as String

### DISCUSSION THREAD

**Discussion by AbhiramYay** [javers/javers#1184] [Q&A] [answered]
Hi,

JaVers is saving entity  **Date** data type field value as **String** in MongoDB **jv_snapshots** collection, Is there a way to store Date data type field value same as Date data type.

I tried using custom JsonTypeAdapter, it didn't work.

We want to automatically delete the jv_snapshots documents based on the created date using MongoDB TTL index feature which requires Date data type field in the snapshots documents.

Please suggest if there is a way to customize to save Date data type value in **jv_snapshots** collection.
Thanks.

**Comment by bartoszwalacik** [comment, marked as answer]
No, there is no such way. Javers internally serializes Snapshots to JSON and persists JSON strings in databases. It's done that way for maximum portability.
